client/state/plugins/installed/actions.js

```diff
@@ -67,8 +69,57 @@ export function activatePlugin( siteId, plugin ) {
 		};
 		dispatch( { ...defaultAction, type: PLUGIN_ACTIVATE_REQUEST } );
 
+		// TODO: Remove when this flux action is completely reduxified
+		Dispatcher.handleViewAction( {
+			type: 'ACTIVATE_PLUGIN',
+			action: 'ACTIVATE_PLUGIN',
+			site,
+			plugin,
+		} );
+
+		const afterActivationCallback = ( data, error ) => {
+			// TODO: Remove when this flux action is completely reduxified
+			Dispatcher.handleServerAction( {
+				type: 'RECEIVE_ACTIVATED_PLUGIN',
+				action: 'ACTIVATE_PLUGIN',
+				site,
+				plugin,
+				data,
+				error,
+			} );
+
+			// Sometime data can be empty or the plugin always
+			// return the active state even when the error is empty.
+			// Activation error is ok, because it means the plugin is already active
+			if (
+				( error && error.error !== 'activation_error' ) ||
+				( ! ( data && data.active ) && ! error )
+			) {
+				dispatch( bumpStat( 'calypso_plugin_activated', 'failed' ) );
+				dispatch(
+					recordTracksEvent( 'calypso_plugin_activated_error', {
+						error: error && error.error ? error.error : 'Undefined activation error',
```

Comment:
I find these `error.error` objects rather confusing. Are there other attributes than `error` and why is it constructed that way? This is more of a general question as it applies to a lot of places in Calypso and is not specific to this PR.

Reply:
While I'm not 100% sure on the top of my head, I think there are other attributes like `statusCode` (the actual HTTP response code) or `message` (the error message), while `error` here is the actual error code as returned by the API. For example, if the API returns a `new WP_Error( 'insufficient_permissions', 'You do not have permissions to access this resource' )`, `message` will be `You do not have permissions to access this resource` and `error` will be `insufficient_permissions`.